src/carousel/carousel.scss

```diff
@@ -0,0 +1,167 @@
+  position: relative;
+}
+
+@mixin iui-carousel-slider {
+  display: flex;
+  position: relative;
+  overflow: hidden;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+
+  &-item {
+    position: absolute;
+    margin-left: auto;
+    margin-right: auto;
+    left: 0;
+    right: 0;
+    flex-shrink: 0;
+    width: 100%;
+    box-sizing: border-box;
+    @media (prefers-reduced-motion: no-preference) {
+      transition: transform $iui-speed-slow ease-in-out;
+    }
+
+    &.iui-previous {
+      transform: translateX(-100%);
+    }
+
+    &.iui-next {
+      transform: translateX(100%);
+    }
+
+    &.iui-current {
+      transform: translateX(0);
+    }
+
+    &-image {
+      width: 100%;
+      height: 100%;
+      background-position: center;
+      background-size: cover;
+    }
+
+    &-title {
+      position: absolute;
+      left: 0;
+      bottom: 0;
+      padding: $iui-baseline $iui-sm;
+      box-sizing: border-box;
+      width: 100%;
+      transition: background-color $iui-speed-fast ease;
+      @include themed {
+        background-color: rgba(0, 0, 0, t(iui-opacity-4));
+        color: t(iui-color-foreground-accessory);
+      }
+
+      &:hover {
+        @include themed {
+          background-color: rgba(0, 0, 0, t(iui-opacity-3));
+        }
+      }
+    }
+  }
+
+  + .iui-carousel-navigation {
+    @include themed {
+      border-top: $iui-xxs solid t(iui-color-background-4);
+    }
+  }
+}
+
+@mixin iui-carousel-navigation {
+  display: flex;
+  align-items: center;
+  height: $iui-baseline * 3;
+
+  &-dots {
+    @include iui-reset;
+    display: flex;
+    flex-grow: 1;
+    white-space: nowrap;
+    overflow: hidden;
+    justify-content: center;
+    align-items: center;
+    list-style: none;
+  }
+
+  &-dot {
+    @include iui-button;
+    @include iui-button-size($size: small);
+    background-color: transparent;
+    outline: none;
+
+    &:hover:not(.iui-invisible):not(.iui-active)::after {
+      @include themed {
+        background-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-3));
+      }
+    }
+
+    &.iui-invisible {
+      cursor: default;
+    }
+
+    &:focus {
+      outline: none;
+
+      &::after {
+        @include themed {
+          outline: 1px solid t(iui-color-foreground-primary);
+          outline-offset: 1px;
+        }
+      }
+    }
+
+    &:focus:not(:focus-visible)::after {
+      @include themed {
+        outline: none;
+      }
+    }
+
+    &::after {
+      content: '';
+      display: block;
+      width: $iui-icons-small;
+      height: $iui-icons-small;
+      border-radius: 50%;
+      transition: background-color $iui-speed-fast ease;
+      @media (prefers-reduced-motion: no-preference) {
+        transition: background-color $iui-speed-fast ease, transform $iui-speed-fast ease;
+      }
+      @include themed {
+        background-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
+      }
+    }
+
+    &.iui-first::after {
```

Comment:
all these not used too in demo, so for now remove?

Reply:
`.iui-first`, `.iui-second`, and `.iui-invisible` would allow us to "truncate" the dots if more than X amount of slide are present.  #374 shows the truncation at the start and end of the dots.  The image below shows what it would look like if you were on slide 1 of 6 or more.
![Screen Shot 2022-01-18 at 4 09 19 PM](https://user-images.githubusercontent.com/849817/150019282-398f808e-e6f0-499d-bb65-7eeca627e7fc.png)

The only reason they are unused in the demo is I don't know how to actually make them react to what slide is current.  I could include a demo to showcase how it's used, but you wouldn't be able to interact with it.

Reply:
> Line 137 is the transition that is applied if the user has reduced motion set while line 138-140 is the transition that is applied if the user has motion enabled.

Technically, they are both applied when motion is enabled, the latter one just takes preference. But yeah that's the idea, and we should document this as a "best practice" somewhere, for future contributors

Reply:
Pushed some JS in 
[`cd5c898`](https://github.com/iTwin/iTwinUI/pull/377/commits/cd5c898c487bdb241a5ea8f3f850c916ec906bc9) to handle dots truncation. It seems to "work" for odd numbers, but looks weird when the selection stays in the middle.

Also temporarily added `display: none` until we figure out the focus situation with invisible dots. W3's example shows that selection follows focus (using left/right arrow keys), so maybe we'll never have to focus invisible dots.
https://www.w3.org/TR/wai-aria-practices-1.2/examples/carousel/carousel-2-tablist.html

Reply:
For me navigation is buggy.. Seems active class is jumping between dots...

Reply:
what do you mean "jumping between dots"?

Reply:
Notice this weird flickering when dots get smaller around
![carousel](https://user-images.githubusercontent.com/81580355/150857187-4432aa56-55d3-4e6c-85cf-e30de69a972d.gif)

Reply:
that's not a bug, it's designed like that and that's exactly what I was talked about when I said "looks weird when the selection stays in the middle".

If there was absolutely no motion, then it would look even more weird because the slide would change but there would be no feedback on the clicked element.

I think we need some animation (maybe using react-transition-group) to show the dots moving

Reply:
@FlyersPh9 had this short video of nice animation, so the only question is, if there is something more/less we need from css to achieve that in react.
But I am resolving this one then..